src/draw/nxp/pxp/lv_pxp_osa.c

```diff
@@ -83,26 +88,23 @@ static pxp_cfg_t _pxp_default_cfg = {
 
 void PXP_IRQHandler(void)
 {
-#if defined(SDK_OS_FREE_RTOS)
-    BaseType_t xHigherPriorityTaskWoken = pdFALSE;
-#endif
-
     if(kPXP_CompleteFlag & PXP_GetStatusFlags(PXP_ID)) {
         PXP_ClearStatusFlags(PXP_ID, kPXP_CompleteFlag);
-#if defined(SDK_OS_FREE_RTOS)
-        xSemaphoreGiveFromISR(xPXPIdleSemaphore, &xHigherPriorityTaskWoken);
-
-        /* If xHigherPriorityTaskWoken is now set to pdTRUE then a context switch
-        should be performed to ensure the interrupt returns directly to the highest
-        priority task.  The macro used for this purpose is dependent on the port in
-        use and may be called portEND_SWITCHING_ISR(). */
-        portYIELD_FROM_ISR(xHigherPriorityTaskWoken);
+#if LV_USE_OS
+        lv_thread_sync_signal_isr(&pxp_sync);
 #else
         ucPXPIdle = true;
 #endif
     }
 }
 
+#if defined(__ZEPHYR__)
+static void PXP_ZephyrIRQHandler(void *)
+{
+    PXP_IRQHandler();
+}
+#endif
```

Comment:
Ah it kinda is - I do not know how many other RTOS give the possibility to add userdata to an ISR.

Reply:
What I mean is the name implies that it's overriding a `weak` definition in the OS like a systick handler that gets called by the OS, but it's really just a local function.

Also, does `PXP_IRQHandler` need to be global? Does it get called by anything external?

Reply:
Ah, no its really just a wrapper to accept the `void *` that the Zephyr impl provides. 
`PXP_IRQHandler` does not need to be global - but that I did not touch. Should I make it a static one as well?

Reply:
Yeah, it's best to not add to the global namespace if it can be avoided. For now, just make `PXP_IRQHandler` static and move this heading above `PXP_IRQHandler` please, and it all looks good :slightly_smiling_face:

```c
/**********************
 *   STATIC FUNCTIONS
 **********************/
```

Reply:
Performed the change and renamed the handler to match the other non namespace scoped functions.